Libraries/src/net/relinc/libraries/sample/TorsionSample.java

```diff
@@ -0,0 +1,197 @@
+		case "outerDiameter":
+			this.setOuterDiameter(Double.parseDouble(val));
+			break;
+		case "length":
+			this.setLength(Double.parseDouble(val));
+			break;
+		}
+	}
+
+	@Override
+	public int addSpecificParametersToDecriptorDictionary(DescriptorDictionary d, int i) {
+		double length;
+		double innerDiameter;
+		double outerDiameter;
+		
+		if(SPSettings.metricMode.get()) {
+			length = SPOperations.round(Converter.mmFromM(this.getLength()), 3);
+			innerDiameter = SPOperations.round(Converter.mmFromM(this.getInnerDiameter()), 3);
+			outerDiameter = SPOperations.round(Converter.mmFromM(this.getOuterDiameter()), 3);
+		} else {
+			length = SPOperations.round(Converter.InchFromMeter(this.getLength()), 3);
+			innerDiameter = SPOperations.round(Converter.InchFromMeter(this.getInnerDiameter()), 3);
+			outerDiameter = SPOperations.round(Converter.InchFromMeter(this.getOuterDiameter()), 3);
+		}
+		
+		d.descriptors.add(i++, new Descriptor("Length", Double.toString(length)));
+		d.descriptors.add(i++, new Descriptor("Inner Diameter", Double.toString(innerDiameter)));
+		d.descriptors.add(i++, new Descriptor("Outer Diameter", Double.toString(outerDiameter)));
+		return i;
+	}
+
+	@Override
+	public String getParametersForPopover(boolean metric) {
+		String common = getCommonParametersForPopover(metric);
+		String des = "";
+		if(metric) {
+			des += "Length: " + SPOperations.round(Converter.mmFromM(length),3) + " mm\n";
+			des += "Inner Diameter: " + SPOperations.round(Converter.mmFromM(this.getInnerDiameter()), 3) + " mm\n";
+			des += "Outer Diameter: " + SPOperations.round(Converter.mmFromM(this.getOuterDiameter()), 3) + " mm\n";
+		} else {
+			des += "Length: " + SPOperations.round(Converter.InchFromMeter(length),3) + " in\n";
+			des += "Inner Diameter: " + SPOperations.round(Converter.InchFromMeter(this.getInnerDiameter()), 3) + " in\n";
+			des += "Outer Diameter: " + SPOperations.round(Converter.InchFromMeter(this.getOuterDiameter()), 3) + " in\n";	
+		}
+		des += common;
+		return des;
+	}
+	
+	public double getStrainRate(double reflectedStrain) {
+		return 4 * (this.getAverageDiameter() / 2) * this.getWavespeed() / (this.barSetup.IncidentBar.diameter / 2 * this.getLength()) * reflectedStrain;
+	}
+	
+	public double getLoad(double barStrain) {
+		return this.getTorque(barStrain) / (this.getAverageDiameter() / 2.0);
+	}
+	
+	private double getTorque(double strain) {
+		return this.getShearModulus() * getPolarMomentOfCylinder(this.barSetup.IncidentBar.getRadius()) * strain / this.barSetup.IncidentBar.getRadius();
+	}
+	
+	private double getStressFromTorque(double torque) {
+		double momentOfInertia = this.getPolarMomentOfTube();
+		return torque / momentOfInertia * ((this.getOuterDiameter() + this.getInnerDiameter()) / 4);
+	}
+	
+	public double getStressFromLoad(double load) {
+		double torque = load * (this.getAverageDiameter() / 2.0);
+		return this.getStressFromTorque(torque);
+	}
+	
+	@Override
+	public double getWavespeed() {
```

Comment:
@bhalonen This is the method you're talking about? This actually uses all parameters from the bar setup if you look at `getShearModulus()`. Perhaps this should be moved to the bar setup class?

Reply:
Ok, will move into bar setup